### PR TITLE
fix: tx chunk store put and delete operation correctness

### LINKED ISSUE #3783: Refactor disk txChunkStore
With the current design of the disk txChunkStore, we have some issues handling transactions that could be operating on the same indexes in the chunkstore. This is because we revert the state of the TxStore in the TxChunkStore and we can have multiple Txns happening at the same time that could be operating on the same indexes. Reverting them directly could cause inconsistencies.

The Txns in storage pkg don't guarantee ACID semantics strictly. The reason this is okay is in most components we will be operating on separate indexes. The only common point would be the ChunkStore. So we need to handle the reverse operations through the ChunkStore API and not look into how it is implemented right now with the Store as part of the transaction.

### PR DESCRIPTION
**Pull request by mrekucci**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3854)

Closes #3783

### CODE REVIEW

File: pkg/localstorev2/internal/chunkstore/transaction.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+	var err *multierror.Error
+	for _, loc := range cs.txSharky.writtenLocs {
+		err = multierror.Append(err, cs.txSharky.Sharky.Release(context.Background(), loc))
+	}
+	return err.ErrorOrNil()
+}
+
+func (cs *txChunkStoreWrapper) NewTx(state *storage.TxState) storage.TxChunkStore {
+	txStore := cs.txStore.NewTx(state)
```
Comment (aloknerurkar) [inline pkg/localstorev2/internal/chunkstore/transaction.go:128]: I thought we decided to remove this transaction?

File: pkg/localstorev2/internal/chunkstore/transaction.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cs *txChunkStoreWrapper) Rollback() error {
+	if err := cs.txStore.Rollback(); err != nil {
```
Comment (aloknerurkar) [inline pkg/localstorev2/internal/chunkstore/transaction.go:116]: What is happening to the `revOps`?

**Review by aloknerurkar — CHANGES_REQUESTED** [review]

File: pkg/localstorev2/internal/chunkstore/transaction.go (hunk 1 above)
Comment (mrekucci) [reply pkg/localstorev2/internal/chunkstore/transaction.go]: Fixed.

File: pkg/localstorev2/internal/chunkstore/transaction.go (hunk 2 above)
Comment (mrekucci) [reply pkg/localstorev2/internal/chunkstore/transaction.go]: Fixed.

File: pkg/localstorev2/internal/chunkstore/transaction.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+	var (
+		sum = sha256.Sum256(buf)
+		loc sharky.Location
+		err error
+	)
+
+	t.opsMu.Lock()
+	defer t.opsMu.Unlock()
+
+	loc, ok := t.toReleaseLocs[sum]
```
Comment (aloknerurkar) [inline pkg/localstorev2/internal/chunkstore/transaction.go:41]: So I cannot think of a better way at the moment. But lets add some comments here on why this is being done.

**Review by aloknerurkar — APPROVED** [review]

File: pkg/localstorev2/internal/chunkstore/transaction.go (hunk 3 above)
Comment (mrekucci) [reply pkg/localstorev2/internal/chunkstore/transaction.go:45]: Done.

**Pull request merged**